src/QMCWaveFunctions/Fermion/MultiSlaterDeterminantFast.cpp

```diff
@@ -962,17 +962,27 @@ void MultiSlaterDeterminantFast::evaluateDerivatives(ParticleSet& P,
     }
   }
 
-  if (Dets.size() != 2)
-  {
-    throw std::runtime_error(
-        "MultiSlaterDeterminantFast::evaluateDerivatives only compatible with two quantum particle types.");
-  }
-  else
+  //Currently, the MultiDiracDeterminant::evaluateDerivatives works with a legacy design, essentially requiring only up and down determinants.
+  //e.g. for spinor cases, we only have one determinant so this interface doesn't work.
+  //Here we throw an error only if the optimization is turned on for MultiDiracDeterminants until the code is updated
+  bool all_Optimizable = true;
+  for (size_t id = 0; id < Dets.size() && all_Optimizable; id++)
```

Comment:
this is a little baroque or perhaps even medieval
```c++
for(auto det : Dets)
  if(!det->Optimizable)
   return;
nameOfDerivativeOnlyCalculatedForSpinSystems();
//or
if(std::all_of(Dets.begin(), Dets.end(), [](auto det){ return det->Optimizable; }))
  nameOfDerivativeOnlyCalculatedForSpinSystems();
```

Reply:
updated.
Had to use (auto cost& det : Dets) since Dets is a vector of unique_ptr. (auto det : Dets) tries to do a copy which fails since they are unique_ptr